### ISSUE TITLE
Support for publicNetworkAccess property on azurerm_container_registry

### ISSUE THREAD

**Issue by adrianmo** [terraform-providers/terraform-provider-azurerm#7334]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The latest API version for the `Microsoft.ContainerRegistry/registries` Azure resource, i.e., `2019-12-01-preview` at the time of writing, has a property called `publicNetworkAccess` to enable or disable public network access to the registry.

The `publicNetworkAccess` property is optional and its type is Enum with allowed values of "Enabled" or "Disabled". The default value is "Enabled".

From the AZ CLI, this property is set via a boolean argument.

```
$ az acr create --help
...
    --public-network-enabled       : Allow public network access for the container registry. The
                                     Default is allowed.  Allowed values: false, true.
...
```

### New or Affected Resource(s)

* azurerm_container_registry

### Potential Terraform Configuration

```hcl
resource "azurerm_container_registry" "acr" {
  name                     = "containerRegistry1"
  resource_group_name      = azurerm_resource_group.rg.name
  location                 = azurerm_resource_group.rg.location
  sku                      = "Premium"
  admin_enabled            = false
  georeplication_locations = ["East US", "West Europe"]
  public_network_enabled   = false
}
```

### References

* https://docs.microsoft.com/en-us/azure/templates/microsoft.containerregistry/registries#registryproperties-object
* https://docs.microsoft.com/en-us/azure/container-registry/container-registry-access-selected-networks#disable-public-network-access

**Comment by masonhuemmer** [comment]
When is this feature going to be available?

**Comment by tombuildsstuff** [comment]
Fixed via #10969
